pwa-devdocs/src/peregrine/reference/talons/CartPage/GiftCards/index.md

```diff
@@ -0,0 +1,92 @@
+---
+title: Gift Cards Talons
+---
+
+<!--
+The reference doc content is generated automatically from the source code.
+To update this section, update the doc blocks in the source code
+-->
+
+{% include auto-generated/peregrine/lib/talons/CartPage/GiftCards/useGiftCards.md %}
+
+## Examples
+
+### useGiftCards()
+
+```jsx
+import React from 'react'
+import { useGiftCards } from '@magento/peregrine/lib/talons/CartPage/GiftCards'
+import {
+    GET_CART_GIFT_CARDS_QUERY,
+    GET_GIFT_CARD_BALANCE_QUERY,
+    APPLY_GIFT_CARD_MUTATION,
+    REMOVE_GIFT_CARD_MUTATION
+} from './myGiftCardQueries';
+
+const MyGiftCards = props => {
+
+ const talonProps = useGiftCards({
+     setIsCartUpdating: props.setIsCartUpdating,
+     mutations: {
+         applyCardMutation: APPLY_GIFT_CARD_MUTATION,
+         removeCardMutation: REMOVE_GIFT_CARD_MUTATION
+     },
+     queries: {
+         appliedCardsQuery: GET_CART_GIFT_CARDS_QUERY,
+         cardBalanceQuery: GET_GIFT_CARD_BALANCE_QUERY
+     }
+ });
+
+ const propsFromTalon = useGiftCards(talonProps)
+
+ const {
+     applyGiftCard,
+     checkBalanceData,
+     checkGiftCardBalance,
+     errorLoadingGiftCards,
+     errorRemovingCard,
+     giftCardsData,
+     isLoadingGiftCards,
+     isApplyingCard,
+     isCheckingBalance,
+     isRemovingCard,
+     removeGiftCard,
+     setFormApi,
+     shouldDisplayCardBalance,
+     shouldDisplayCardError
+ } = propsFromTalon;
```

Comment:
This seems like a lot of duplication and brittleness in each one of these 😬

Reply:
Surprised we aren't just including this as the `@example`. I wouldn't want to have to maintain these.

Reply:
@supernova-at 
Code samples are something the community always request more of, so I've tried to add more value to these reference docs by providing simple examples.
Any code samples we provide has the potential to become out of date at some point in the future, but the community is pretty good at letting us know when this happens.

@sirugh 
I thought about putting these in the source code inside `@example` tags, but the size of some of these examples can balloon the file content greatly.

Reply:
Fair. I just wish we had a "living" doc like storybook.